dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractHostedEntryResource.java

```diff
@@ -184,6 +184,8 @@ public T editHosted(@ApiParam(hidden = true) @Auth User user,
 
         checkVersionLimit(user, entry);
 
+        // Ensure all paths are absolute (copy path to absolute path)
```

Comment:
Don't think this is quite right either. 
For files for non-hosted workflows, the path is relative. The absolute path is well, absolute. 
If hosted workflows have absolute paths in both places, that may cause problems too.

Reply:
The issue is that we don't actually store these hosted files in a directory structure, it is purely a string. We do need the absolute path so that when we download a hosted entry version it creates the proper directory structure. We cannot necessarily go off of the primary descriptor for this.

Maybe a solution is to set the absolute path during validation, though this makes the assumption that all the files given are imported somehow. A secondary file could be added even if the primary never links to it.

Reply:
A solution could be that during the parsing of the primary descriptor, we set the absolute path in a similar way to when we do refresh of a remote workflow/tool. This won't work for any file that is not reference in the files related to the primary descriptor, though maybe that is ok.

Reply:
After looking I don't this this is easy. Maybe we should have all files relative to root. Since this is more like a gist than a repo, I think it is ok.